### DISCUSSION TITLE
What is the recommend memory space for the backend database?

### DISCUSSION THREAD

**Discussion by hailindu** [apache/airflow#21437] [Q&A] [answered]
My team is planning to create a cloud-based backend database to store the metadata from airflow, but we are not clear on how big the database should be. 1GB, 4GB?

I do know that this is based on how heavily we use airflow, but is there any documentation that I can review as a reference or any related doc that I can review to estimate the backend database memory size?

**Comment by potiuk** [comment, marked as answer]
It depend on many factors. The "quick-start" https://airflow.apache.org/docs/apache-airflow/stable/start/docker.html mentions 4GB RAM for quick start - but this involves DockerComppose standard configuration we provided and tested. 

For your installation, you are responsible to test and asses the requirements. Airflow team is not able to asses all the parameters of your implementation. Each of the components you run can run aribitrary number of DAGs, files, tasks, queries that we simply are not able to asses. Each of those might arbitrary increase the requirements. Just to give you an example - if you have a dag that create an object that takes 3GB of RAM - you will have to accomodate for it. It happens. It depends.

We recommend you to run your own assesment and run your own benchmarking to come to the right sizing for your case.
BTW. Would you trust any value we provide you ? What happens if we say (for example)  "3GB is enough" (which we can't) ? 
Would you sue us? 
You are really responsible to size your system and any answer from our side is not making you less responsible?

**Reply by potiuk** [reply]
Quick answer: Just get example dags that you are going to run, test how much memory they need, extrapolate with the number of DAGs you have and guessed complexiyt and you get your estimate.

**Reply by hailindu** [reply]
hi Potiuk, thank you so much for confirming it. That def sounds like a reasonable approach
